config/v1/types_console.go

```diff
@@ -26,6 +29,26 @@ type Console struct {
 type ConsoleSpec struct {
 	// +optional
 	Authentication ConsoleAuthentication `json:"authentication"`
+	// route contains hostname and serving certificate for replacing the default hostname,
+	// under which is console available.
+	// If not specified, default route will be created by the console-operator.
+	// +optional
+	Route CustomRoute `json:"route,omitempty"`
+}
+
+// CustomRoute replaces the default console hostname under which console will be available.
+type CustomRoute struct {
+	// hostname is the desired custom address under which console will be available.
+	Hostname string `json:hostname`
```

Comment:
Here I wasn't sure if we dont wanna follow the Route naming, which would be `Host`

Reply:
I think we should probably create `ConsoleConfigRoute` and do the same as https://github.com/openshift/api/blob/master/imageregistry/v1/types.go#L302.  Three fields, include `Name`.  Consistency is good. 

Provided we don't go the central config route that is currently being discussed.

Reply:
```golang
// ConsoleConfigRoute holds information on external route access to console
type ConsoleConfigRoute struct {
	// name of the route to be created.
	Name string `json:"name" protobuf:"bytes,1,opt,name=name"`
	// hostname for the route.
	// +optional
	Hostname string `json:"hostname,omitempty" protobuf:"bytes,2,opt,name=hostname"`
	// secretName points to secret containing the certificates to be used
	// by the route.
	// +optional
	SecretName string `json:"secretName,omitempty" protobuf:"bytes,3,opt,name=secretName"`
}
```